snakemake_storage_plugin_irods/__init__.py

```
@@ -453,10 +307,65 @@ def list_candidate_matches(self) -> Iterable[str]:
+
+
+def _next_wildcard(input_str: str) -> tuple[str, Optional[str]]:
+    next_index = input_str.find("{")
+    if next_index == -1:
+        return input_str, None
+    if next_index < len(input_str)-1 and input_str[next_index+1] == "{":
+        new_input_str, remaining_str = _next_wildcard(input_str[next_index+2:])
+        return input_str[:next_index+2] + new_input_str, remaining_str
+    last_index = input_str.find("}")
+    if last_index == -1:
+        raise ValueError(f"Bracket not closed properly in: {input_str}")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)